### PR TITLE
Java: Add `Zremrangebyscore` command. (Sorted Set Group)

### PR DESCRIPTION
**Pull request by SanHalacogluImproving**

### CODE REVIEW

File: java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java [hunk 1]
```diff
@@ -456,4 +459,31 @@ CompletableFuture<Map<String, Double>> zrangeWithScores(
...
+     *     If <code>key</code> does not exist, it is treated as an empty sorted set, and the command
+     *     returns <code>0</code>.<br>
+     *     If <code>minScore</code> is greater than <code>maxScore</code>, <code>0</code> is returned.
+     * @example
+     *     <pre>{@code
+     * Long payload1 = client.zremrangebyscore("mySortedSet", new ScoreBoundary(1, false), new ScoreBoundary(5)).get();
+     * assert payload1 == 4L; // Indicates that 4 members, with scores ranging from 1 (exclusive) to 5 (inclusive), have been removed from "mySortedSet".
+     *
+     * Long payload2 = client.zremrangebyscore("nonExistingSortedSet", InfScoreBound.NEGATIVE_INFINITY , InfScoreBound.POSITIVE_INFINITY).get();
+     * assert payload2 == 0;
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java:485]: This example is more intersting
```suggestion
     * Long payload2 = client.zremrangebyscore("mySortedSet", NEGATIVE_INFINITY, -42).get();
     * assert payload2 == 0; // Indicates that no elements were removed
```

File: java/client/src/test/java/glide/api/models/TransactionTests.java [hunk 2]
```diff
@@ -427,15 +427,12 @@ public void transaction_builds_protobuf_request(BaseTransaction<?> transaction)
         transaction.zrank("key", "member");
         results.add(Pair.of(Zrank, ArgsArray.newBuilder().addArgs("key").addArgs("member").build()));
 
-        transaction.zrankWithScore("key", "member");
+        transaction.zremrangebyscore(
```
Comment (Yury-Fridlyand) [inline java/client/src/test/java/glide/api/models/TransactionTests.java:430]: don't remove existing test

Comment (SanHalacogluImproving) [reply java/client/src/test/java/glide/api/models/TransactionTests.java:441]: ooops fixed

**Review by Yury-Fridlyand — APPROVED** [review]

**Comment by Yury-Fridlyand** [comment]
CI is red - spotless?

File: java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java [hunk 3]
```diff
@@ -456,4 +459,31 @@ CompletableFuture<Map<String, Double>> zrangeWithScores(
...
+     * Removes all elements in the sorted set stored at <code>key</code> with a score between <code>
+     * minScore</code> and <code>maxScore</code>.
+     *
+     * @see <a href="https://redis.io/commands/zremrangebyscore/">redis.io</a> for more details.
+     * @param key The key of the sorted set.
+     * @param minScore The minimum score to remove from. Can be an implementation of {@link
+     *     InfScoreBound} representing positive/negative infinity, or {@link ScoreBoundary}
+     *     representing a specific score and inclusivity.
+     * @param maxScore The maximum score to remove to. Can be an implementation of {@link
+     *     InfScoreBound} representing positive/negative infinity, or {@link ScoreBoundary}
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java:473]:
```suggestion
     *     InfScoreBound} to specify a positive/negative infinity bound, or {@link ScoreBoundary}
```

File: java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java [hunk 4]
```diff
@@ -456,4 +459,31 @@ CompletableFuture<Map<String, Double>> zrangeWithScores(
...
+     *     representing a specific score and inclusivity.
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java:474]:
```suggestion
     *     to specify a specific score and inclusivity bound.
```

File: java/integTest/src/test/java/glide/SharedCommandTests.java [hunk 5]
```diff
@@ -1249,6 +1249,41 @@ public void zrank(BaseClient client) {
...
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void zremrangebyscore(BaseClient client) {
+        String key1 = UUID.randomUUID().toString();
+        String key2 = UUID.randomUUID().toString();
+        Map<String, Double> membersScores = Map.of("one", 1.0, "two", 2.0, "three", 3.0, "four", 4.0);
+        assertEquals(4, client.zadd(key1, membersScores).get());
+
+        assertEquals(
+                2, client.zremrangebyscore(key1, new ScoreBoundary(1, false), new ScoreBoundary(3)).get());
```
Comment (acarbonetto) [inline java/integTest/src/test/java/glide/SharedCommandTests.java:1262]: Can we rename `ScoreBoundary` to `ScoreBound` to be consistent with `InfScoreBound`?

File: java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java [hunk 6]
```diff
@@ -456,4 +459,31 @@ CompletableFuture<Map<String, Double>> zrangeWithScores(
...
      */
     CompletableFuture<Object[]> zrankWithScore(String key, String member);
+
+    /**
+     * Removes all elements in the sorted set stored at <code>key</code> with a score between <code>
+     * minScore</code> and <code>maxScore</code>.
+     *
+     * @see <a href="https://redis.io/commands/zremrangebyscore/">redis.io</a> for more details.
+     * @param key The key of the sorted set.
+     * @param minScore The minimum score to remove from. Can be an implementation of {@link
```
[the remark this answers is not in the document]
Comment (SanHalacogluImproving) [reply java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java:469]: These are taken for Node and PY do you want to change them? Seems all minor to me.

File: java/integTest/src/test/java/glide/SharedCommandTests.java (hunk 5 above)
Comment (SanHalacogluImproving) [reply java/integTest/src/test/java/glide/SharedCommandTests.java:1262]: Sure do you want that in another PR?

Comment (acarbonetto) [reply java/integTest/src/test/java/glide/SharedCommandTests.java:1262]: You can do this here.

Comment (SanHalacogluImproving) [reply java/integTest/src/test/java/glide/SharedCommandTests.java:1262]: I'll do it another PR will be much cleaner for review.

**Pull request merged**